activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ModelServiceImpl.java

```diff
@@ -90,21 +93,28 @@
 
     private final HashMap<String, String> modelIdentifiers = new HashMap();
 
+    private final Map<String, List<ModelUpdateListener>> modelUpdateListenersMapByModelType;
+
     @Autowired
     public ModelServiceImpl(ModelRepository modelRepository,
                             ModelTypeService modelTypeService,
                             ModelContentService modelContentService,
                             ModelExtensionsService modelExtensionsService,
                             JsonConverter<Model> jsonConverter,
-                            ProcessModelContentConverter processModelContentConverter) {
+                            ProcessModelContentConverter processModelContentConverter,
+                            Set<ModelUpdateListener> modelUpdateListeners) {
```

Comment:
Wouldn't it be better to use `List` here instead of `Set` as the type used by `modelUpdateListenersMapByModelType ` is `List`?

Reply:
I have changed the `modelUpdateListenersMapByModelType` to be a map of string, set as I prefer the `modelUpdateListeners`to be a set and avoid duplications in the registration of listeners.

Reply:
We need it to be a set to avoid duplications in the listener registering and for Spring context to create an empty set if no listeners are registered. 
This is the same implementation followed in `ContentUpdateListener` registry pattern.